### PR TITLE
added flux model

### PR DESCRIPTION
**Pull request by dsingal0**
added flux schnell and flux dev

### CODE REVIEW

File: flux/dev/config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+external_package_dirs: []
+model_metadata:
+  example_model_input: {"prompt": 'black forest gateau cake spelling out the words "FLUX DEV", tasty, food photography, dynamic shot'}
```
Comment (vshulman) [inline flux/dev/config.yaml:3]: thanks for adding this

File: flux/dev/model/model.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+class Model:
+    def __init__(self, **kwargs):
+        self._secrets = kwargs["secrets"]
+        self.model_name = kwargs["config"]["model_metadata"]["repo_id"]
+        self.hf_access_token = self._secrets["hf_access_token"]
+        self.pipe = None
+
+    def load(self):
+        self.pipe = FluxPipeline.from_pretrained(
+            self.model_name, torch_dtype=torch.bfloat16, token=self.hf_access_token
```
Comment (vshulman) [inline flux/dev/model/model.py:27]: nit: `torch_dtype` auto might be faster?

File: flux/dev/model/model.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            result = subprocess.run(
+                ["nvidia-smi"], capture_output=True, text=True, check=True
+            )
+            logging.info(result.stdout)
+        except subprocess.CalledProcessError as e:
+            logging.error(f"Command failed with code {e.returncode}: {e.stderr}")
+
+    def convert_to_b64(self, image: Image) -> str:
+        buffered = BytesIO()
+        image.save(buffered, format="WEBP")  # webp is more efficient than jpeg
```
Comment (vshulman) [inline flux/dev/model/model.py:40]: but then you get a webp file

File: flux/dev/model/model.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        width = model_input.get("width", 1024)
+        height = model_input.get("height", 1024)
+        if not seed:
+            seed = random.randint(0, MAX_SEED)
+        if len(prompt) > max_sequence_length:
+            logging.warning(
+                f"Input longer than {max_sequence_length} tokens, truncating"
+            )
+            prompt = prompt[:max_sequence_length]
+        if prompt2 and len(prompt2) > max_sequence_length:
```
Comment (vshulman) [inline flux/dev/model/model.py:63]: no need to do the same warning?

**Review by vshulman — CHANGES_REQUESTED** [review]
let's copy a readme from one of the other image models, in particular explaining how to use the b64 encoding?

File: flux/dev/config.yaml (hunk 1 above)
Comment (dsingal0) [reply flux/dev/config.yaml:3]: 👍 I think its in the CONTRIBUTING.md as a requirement lol

File: flux/dev/model/model.py (hunk 2 above)
Comment (dsingal0) [reply flux/dev/model/model.py:27]: auto is supposed to infer the dtype from the weights, but in their calling code, flux explicitly define bfloat16, so I used that.
https://huggingface.co/black-forest-labs/FLUX.1-dev

File: flux/dev/model/model.py (hunk 4 above)
Comment (dsingal0) [reply flux/dev/model/model.py:63]: should add the same warning. will do

File: flux/README.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+# Function used to convert a base64 string to a PIL image
+def b64_to_pil(b64_str):
+    return Image.open(BytesIO(base64.b64decode(b64_str)))
+
+data = {
+  "prompt": "a little boy looking through a large magical portal, the boy sees a futuristic human civilization in that portal, extremely detailed, trending on artstation, 8k"
+}
+
+# Call model endpoint
+res = httpx.post(
```
Comment (vshulman) [inline flux/README.md:56]: <3

File: flux/README.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    f"https://model-{model_id}.api.baseten.co/production/predict",
```
Comment (vshulman) [inline flux/README.md:57]: nit: your truss push is not `--publish` then the endpoints would be development. Maybe update it to `publish` above?

**Review by vshulman — APPROVED** [review]
see note about development vs prod in the readme

File: flux/README.md (hunk 6 above)
Comment (dsingal0) [reply flux/README.md:57]: I see, good catch.

**Pull request merged**